File: sample/src/main/kotlin/com/vicpin/cleanrecyclerview/sample/view/presenter/AdapterItemPresenter.java
package com.vicpin.cleanrecyclerview.sample.view.presenter;

import com.vicpin.cleanrecyclerview.sample.model.Item;
import com.vicpin.presenteradapter.ViewHolderPresenter;

/**
 * Created by Alvaro on 21/12/2016.
 */

public class AdapterItemPresenter extends ViewHolderPresenter<Item, AdapterItemPresenter.View> {

    @Override public void onCreate() {
        setContent();
    }

    private void setContent() {
        getView().setTitle(getData().getTitle());
        getView().setDescription(getData().getDescription());
        getView().setImage(getData().getImageUrl());
    }

    public interface View {
        void setTitle(String title);

        void setDescription(String description);

        void setImage(String url);
    }
}

File: sample/src/main/kotlin/com/vicpin/cleanrecycler/sample/view/presenter/AdapterItemPresenter.kt
package com.vicpin.cleanrecycler.sample.view.presenter

import com.vicpin.cleanrecycler.sample.model.Item
import com.vicpin.kpresenteradapter.ViewHolderPresenter

/**
 * Created by Alvaro on 21/12/2016.
 */

class AdapterItemPresenter : ViewHolderPresenter<Item, AdapterItemPresenter.View>() {

    override fun onCreate() {
        setContent()
    }

    private fun setContent() {
        view?.setTitle(data.title)
        view?.setDescription(data.description)
        view?.setImage(data.imageUrl)
    }

    interface View {
        fun setTitle(title: String)

        fun setDescription(description: String)

        fun setImage(url: String)
    }
}
